pypdf/_writer.py

```diff
@@ -2421,17 +2421,37 @@ def pageMode(self, mode: PagemodeType) -> None:  # deprecated
         deprecation_with_replacement("pageMode", "page_mode", "3.0.0")
         self.page_mode = mode
 
-    def add_annotation(self, page_number: int, annotation: Dict[str, Any]) -> None:
+    def add_annotation(
+        self,
+        page_number: Union[int, PageObject],
+        annotation: Dict[str, Any],
+    ) -> DictionaryObject:
+        """
+        Add a single annotation to the page. Must be a new annotation (can not be recycled)
+
+        Args:
+            page_number: page object or number
+            annotation : annotation to be added (created with annotation)
+
+        Returns:
+            the inserted object (to be used in pop-up creation argument for example)
```

Comment:
```suggestion
        Add a single annotation to the page.
        
        The added annotation must be a new annotation.
        It can not be recycled.

        Args:
            page_number: PageObject or page index.
            annotation: Annotation to be added (created with annotation).

        Returns:
            The inserted object
            This can be used for pop-up creation, for example
```